### PR TITLE
Add piece class and implementation

### PR DESCRIPTION
**Pull request by jajsmith**

### CODE REVIEW

**Comment by jajsmith** [comment]
Instead of pointing back to the Cell that we're in, why don't we just accept both the starting and ending coordinates, and then we can bypass knowing the cell and changing the cell each time. Instead when we move a piece, the only thing that needs to change is which cell points to the piece (rather than both ways).

File: piece.h [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	int worth;
+	// A single character name to represent the piece on the board
+	char name;
+
+  public:
+  	// Constructor that changes capitalization of the name based on player
+  	Piece(int player, char name);
+
+  	// Allow game to be set
+  	void setGame(Game game);
```
Comment (dmamujee) [inline piece.h]: should this be a pointer to a game as the parameter?

File: piece.h [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  	Game getGame();
```
Comment (dmamujee) [inline piece.h]: And do you think it would work better to return Game pointer?

File: piece.h (hunk 1 above)
Comment (jajsmith) [inline piece.h]: And it should be const as well

File: piece.cc [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+// ========================
+// Getters & Setters
+// ========================
+
+Piece::setGame(Game game) {
+	this->game = game;
+}
+
+Game Piece::getGame() {
+	return game;
```
Comment (dmamujee) [inline piece.cc:24]: I don't think the types match here, as game is a Game*

**Comment by dmamujee** [comment]
looks good, except the type mismatch for getGame()

**Comment by jajsmith** [comment]
Haha so right!

**Comment by dmamujee** [comment]
I must have missed the message about omitting the pointer back to the Cell, and I agree with that!

**Pull request merged**